File: examples/crypto-tracker/sdks/go/services.go
package sdk

import (
    "context"
)

// CoinGeckoService provides access to 6 API endpoints.
type CoinGeckoService struct {
    Client *Client
}

// NewCoinGeckoService initializes a new CoinGeckoService
func NewCoinGeckoService(client *Client) *CoinGeckoService {
    return &CoinGeckoService{Client: client}
}

// GetCoinDetail: GET coins/{id}
func (s *CoinGeckoService) GetCoinDetail(ctx context.Context, id string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "id": id,
    }
    return s.Client.DoRequest(ctx, "GET", "coins/{id}", pathParams, queryParams, nil, nil)
}

// GetGlobalData: GET global
func (s *CoinGeckoService) GetGlobalData(ctx context.Context) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "global", pathParams, nil, nil, nil)
}

// GetMarketChart: GET coins/{id}/market_chart
func (s *CoinGeckoService) GetMarketChart(ctx context.Context, id string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "id": id,
    }
    return s.Client.DoRequest(ctx, "GET", "coins/{id}/market_chart", pathParams, queryParams, nil, nil)
}

// GetTopCoins: GET coins/markets
func (s *CoinGeckoService) GetTopCoins(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "coins/markets", pathParams, queryParams, nil, nil)
}

// GetTrendingCoins: GET search/trending
func (s *CoinGeckoService) GetTrendingCoins(ctx context.Context) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "search/trending", pathParams, nil, nil, nil)
}

// SearchCoins: GET search
func (s *CoinGeckoService) SearchCoins(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "search", pathParams, queryParams, nil, nil)
}

File: examples/crypto-tracker/sdks/java/Services.java
package com.app.sdk;

import java.util.HashMap;
import java.util.Map;

public final class Services {
    private Services() {}

    public static class CoinGeckoService {
        private final Client client;

        public CoinGeckoService(Client client) {
            this.client = client != null ? client : new Client();
        }

        public CoinGeckoService() {
            this(new Client());
        }

        /** GET coins/{id} */
        public ApiResponse<String> getCoinDetail(String id, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("id", id);
            return this.client.send("GET", "coins/{id}", pathParams, queryParams, null, null);
        }

        /** GET global */
        public ApiResponse<String> getGlobalData() {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "global", pathParams, null, null, null);
        }

        /** GET coins/{id}/market_chart */
        public ApiResponse<String> getMarketChart(String id, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("id", id);
            return this.client.send("GET", "coins/{id}/market_chart", pathParams, queryParams, null, null);
        }

        /** GET coins/markets */
        public ApiResponse<String> getTopCoins(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "coins/markets", pathParams, queryParams, null, null);
        }

        /** GET search/trending */
        public ApiResponse<String> getTrendingCoins() {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "search/trending", pathParams, null, null, null);
        }

        /** GET search */
        public ApiResponse<String> searchCoins(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "search", pathParams, queryParams, null, null);
        }

    }

}